src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -140,6 +144,14 @@ export class WorkspaceSavedObjectsClientWrapper {
     }
   }
 
+  private isDashboardAdmin(request: OpenSearchDashboardsRequest): boolean {
+    const config = this.config || ({} as ConfigSchema);
```

Comment:
Looking at this piece of code, it seems there is a possibility that when `isDashboardAdmin` is called, but `this.config` hasn't initialized, in that case, `isDashboardAdmin` will return false which causes the client wrapper to return incorrect client functions if the current user is admin. Correct me if I'm wrong :)

Reply:
I figured it out, the `isDashboardAdmin` check is run on each request, then I think it will be fine.

Reply:
Yeah, the wrapper will be called in every request(though I think it is a waste of computing capability) so we can return back the right saved objects client according to backend roles of current requesting user.